### ISSUE TITLE
Different rule configuration for annotated method

### ISSUE THREAD

**Issue by FelixZY** [detekt/detekt#7893]
I'm configuring detekt in an existing android project with compose. While most issues have been addressed, there are still a lot of `LongMethod` warnings. I think the default limit of 60 lines make sense for regular code. However, in `@Composable` methods, this is a bit restrictive and I think it would make sense to allow e.g. 100 lines instead.

I [found that detekt supports *suppressors*](https://slack-chats.kotlinlang.org/t/23452872/hi-guys-do-you-think-it-makes-sense-to-skip-the-longmethod-r) that can disable a rule for methods with a given annotation. However, in my case, I would like a *different* rule configuration for methods with a given annotation. Perhaps something like

```yml
complexity:
  LongMethod:
    - threshold: 60
    - threshold: 100
      filterAnnotated:
      - Composable
```

**Comment by cortinico** [comment]
> However, in @Composable methods, this is a bit restrictive and I think it would make sense to allow e.g. 100 lines instead.

This will be possible in detekt 2.0 with this:
- https://github.com/detekt/detekt/issues/6738

Closing as this is practically already implemented, we just need to ship 2.0
